src/sinks/datadog/metrics/config.rs

```diff
@@ -96,7 +96,23 @@ impl DatadogMetricsEndpoint {
 
     // Creates an instance of the `Series` variant with the default API version.
     pub fn series() -> Self {
-        Self::Series(SeriesApiVersion::get_api_version_backwards_compatible())
+        Self::Series(SeriesApiVersion::get_api_version())
+    }
+
+    pub(super) const fn payload_limits(self) -> DatadogMetricsPayloadLimits {
+        // from https://docs.datadoghq.com/api/latest/metrics/#submit-metrics
+        match self {
+            // Sketches use the same payload size limits as v1 series
+            DatadogMetricsEndpoint::Series(SeriesApiVersion::V1)
+            | DatadogMetricsEndpoint::Sketches => DatadogMetricsPayloadLimits {
+                uncompressed: 62_914_560, // 62 MB
+                compressed: 3_200_000,    // 3.2 MB
+            },
+            DatadogMetricsEndpoint::Series(SeriesApiVersion::V2) => DatadogMetricsPayloadLimits {
+                uncompressed: 5_242_880, // 5 MB
+                compressed: 512_000,     // 500 KB
```

Comment:
There's a bit of unit confusion here. These four values are, respectively, 60 MiB, 3.2 MB, 5 MiB, and either 500 KiB or 512 KB. Note that the first number is just wrong either way, but the switching between MiB and MB is awkward.

Reply:
👍 I'll fix that up but ima partially blame https://docs.datadoghq.com/api/latest/metrics/#submit-metrics-v2 .


v2:

```
"The maximum payload size is 500 kilobytes (512000 bytes). Compressed payloads must have a decompressed size of less than 5 megabytes (5242880 bytes)."
```

v1:

```
"The maximum payload size is 3.2 megabytes (3200000 bytes). Compressed payloads must have a decompressed size of less than 62 megabytes (62914560 bytes)."
```

> but the switching between MiB and MB is awkward.

Im not disagreeing but I wonder if that has something to do with the uncompressed vs compressed nature of things.

Reply:
I doubt it. It's not consistent either way.